src/test/java/com/parkhomovsky/bookstore/services/BookServiceTest.java

```diff
@@ -0,0 +1,206 @@
+            .setAuthor("U.N. LoLovich")
+            .setIsbn("978-0-9767736-6-5")
+            .setPrice(new BigDecimal("25"))
+            .setDescription("Book about why you shoudn`t play Dota 2."
+                    + " And why you can`t still find a girl.")
+            .setCoverImage("resources/bookImages/Dota-Pain.png")
+            .setCategories(Set.of(COMEDY_CATEGORY));
+    private static final BookDto RESPONSE_CREATED_DOTA_BOOK_DTO = new BookDto()
+            .setId(2L)
+            .setTitle("Dota 2: Pain and Casino")
+            .setAuthor("U.N. LoLovich")
+            .setIsbn("978-0-9767736-6-5")
+            .setPrice(new BigDecimal("25"))
+            .setDescription("Book about why you shoudn`t play Dota 2."
+                    + " And why you can`t still find a girl.")
+            .setCoverImage("resources/bookImages/Dota-Pain.png")
+            .setCategoryIds(Set.of(2L));
+    private static final Book EXIST_TEST_BOOK = new Book()
+            .setTitle("Test Book")
+            .setAuthor("Test Author")
+            .setIsbn("1315616")
+            .setPrice(new BigDecimal("14.56"))
+            .setDescription("Test book description")
+            .setCoverImage("URL")
+            .setCategories(Set.of(FICTION_CATEGORY));
+    private static final BookDto RESPONSE_EXIST_TEST_BOOK_DTO = new BookDto()
+            .setId(1L)
+            .setTitle("Test Book")
+            .setAuthor("Test Author")
+            .setIsbn("1315616")
+            .setPrice(new BigDecimal("14.56"))
+            .setDescription("Test book description")
+            .setCoverImage("URL")
+            .setCategoryIds(Set.of(1L));
+    private static final BookDtoWithoutCategoryIds
+            RESPONSE_EXIST_TEST_BOOK_DTO_WITHOUT_CATEGORIES = new BookDtoWithoutCategoryIds()
+            .setId(1L)
+            .setTitle("Test Book")
+            .setAuthor("Test Author")
+            .setIsbn("1315616")
+            .setPrice(new BigDecimal("14.56"))
+            .setDescription("Test book description")
+            .setCoverImage("URL");
+    private static final List<BookDtoWithoutCategoryIds> ALL_FICTION_BOOKS_DTO
+            = List.of(RESPONSE_EXIST_TEST_BOOK_DTO_WITHOUT_CATEGORIES);
+    private static final Long ABSENT_BOOK_ID = 128L;
+
+    private static final List<BookDto> ALL_BOOKS_DTO_RESPONSE =
+            List.of(RESPONSE_EXIST_TEST_BOOK_DTO, RESPONSE_CREATED_DOTA_BOOK_DTO);
+    private static final List<Book> ALL_BOOKS = List.of(EXIST_TEST_BOOK, CREATED_DOTA_BOOK);
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Verify successful creating book with a valid input")
+    public void create_validBookParameters_shouldReturnBookDto() {
```

Comment:
This is optional, because naming differs on projects, but in our lessons were shown this style: 
![image](https://github.com/CodeWebWeaver/bookstore/assets/70883268/33b14e63-121c-4403-aab1-667ffcac6667)

Reply:
I did`nt catch your request this method corresponds to our naming style